test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1304 @@
+			var expectedCPULimitString, expectedMemLimitString string
+			expectedMemLimitInBytes := tc.Resources.Limits.Memory().Value()
+			cpuRequest := tc.Resources.Requests.Cpu()
+			cpuLimit := tc.Resources.Limits.Cpu()
+			if cpuRequest.IsZero() && !cpuLimit.IsZero() {
+				expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuLimit.MilliValue()))
+			} else {
+				expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
+			}
+			cpuQuota := kubecm.MilliCPUToQuota(cpuLimit.MilliValue(), kubecm.QuotaPeriod)
+			if cpuLimit.IsZero() {
+				cpuQuota = -1
+			}
+			expectedCPULimitString = strconv.FormatInt(cpuQuota, 10)
+			expectedMemLimitString = strconv.FormatInt(expectedMemLimitInBytes, 10)
+			if podOnCgroupv2Node {
+				if expectedCPULimitString == "-1" {
+					expectedCPULimitString = "max"
+				}
+				expectedCPULimitString = fmt.Sprintf("%s %s", expectedCPULimitString, CPUPeriod)
+				if expectedMemLimitString == "0" {
+					expectedMemLimitString = "max"
+				}
+				// convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight value
+				// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
+				expectedCPUShares = int64(1 + ((expectedCPUShares-2)*9999)/262142)
+			}
+			if expectedMemLimitString != "0" {
+				err := verifyCgroupValue(ci.Name, cgroupMemLimit, expectedMemLimitString)
+				if err != nil {
+					return err
+				}
+			}
+			err := verifyCgroupValue(ci.Name, cgroupCPULimit, expectedCPULimitString)
+			if err != nil {
+				return err
+			}
+			err = verifyCgroupValue(ci.Name, cgroupCPURequest, strconv.FormatInt(expectedCPUShares, 10))
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func waitForContainerRestart(ctx context.Context, f *framework.Framework, podClient *e2epod.PodClient, pod *v1.Pod, expectedContainers []TestContainerInfo) error {
+	var restartContainersExpected []string
+	for _, ci := range expectedContainers {
+		if ci.RestartCount > 0 {
+			restartContainersExpected = append(restartContainersExpected, ci.Name)
+		}
+	}
+	if len(restartContainersExpected) == 0 {
+		return nil
+	}
+	pod, err := podClient.Get(ctx, pod.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	restartedContainersCount := 0
+	for _, cName := range restartContainersExpected {
+		cs, _ := podutil.GetContainerStatus(pod.Status.ContainerStatuses, cName)
+		if cs.RestartCount < 1 {
+			break
+		}
+		restartedContainersCount++
+	}
+	if restartedContainersCount == len(restartContainersExpected) {
+		return nil
+	}
+	if restartedContainersCount > len(restartContainersExpected) {
+		return fmt.Errorf("more container restarts than expected")
+	} else {
+		return fmt.Errorf("less container restarts than expected")
+	}
+}
+
+func waitForPodResizeActuation(ctx context.Context, f *framework.Framework, c clientset.Interface, podClient *e2epod.PodClient, pod, patchedPod *v1.Pod, expectedContainers []TestContainerInfo) *v1.Pod {
```

Comment:
ditto let's add GingkoHelper()

Reply:
Will try and comeback with a commit GinkgoHelper, thanks for sharing

Reply:
Done